libs/template/renderer_test.go

```diff
@@ -189,6 +189,18 @@ My email is {{template "email"}}
 	assert.Contains(t, statement, `My email is hrithik.roshan@databricks.com`)
 }
 
+func TestRendererExecuteTemplateWithUnknownProperty(t *testing.T) {
+	templateText := `{{.does_not_exist}}`
+
+	r := renderer{
+		config:       map[string]any{},
+		baseTemplate: template.New("base"),
+	}
+
+	_, err := r.executeTemplate(templateText)
+	assert.ErrorContains(t, err, "map has no entry for key \"does_not_exist\"")
```

Comment:
The error message will be confusing to the uninitiated user. I would expect something like: `variable "does_not_exist" not defined`. You can check if this error has a particular type and convert it if so.

Reply:
The error does not have a particular type. Fair point. I'll add some regex parsing.

Reply:
@shreyas-goenka Its type is: https://pkg.go.dev/text/template#ExecError, but pretty generic so still need regexp.

Reply:
Yeah, exactly.

Reply:
done